src/functionality.js

```diff
@@ -0,0 +1,118 @@
+/* eslint-disable no-use-before-define */
+const taskList = document.getElementById('task-list');
+
+let tasks = JSON.parse(localStorage.getItem('tasks')) || [];
+
+export function addTask(description) {
+  const task = {
+    description,
+    completed: false,
+    index: tasks.length + 1,
+  };
+  tasks.push(task);
+  renderTask(task);
+  saveTasks();
+}
+
+export function renderTask(task) {
+  const li = document.createElement('li');
+  li.innerHTML = `
+    <input type="checkbox" class="checkbox" ${task.completed ? 'checked' : ''}>
+    <span class="task-description ${task.completed ? 'completed' : ''}">${
+  task.description
+}</span>
+    <span class="edit-task">Edit</span>
+    <span class="delete-task">X</span>
+`;
+  li.classList.add('task');
+  li.index = task.index;
+  taskList.appendChild(li);
+
+  const checkbox = li.querySelector('.checkbox');
+  checkbox.addEventListener('change', (e) => {
+    const { index } = li;
+    tasks[index - 1].completed = e.target.checked;
+    li.querySelector('.task-description').classList.toggle(
+      'completed',
+      e.target.checked,
+    );
+    saveTasks();
+  });
+
+  const taskDescription = li.querySelector('.task-description');
+  taskDescription.addEventListener('blur', (e) => {
+    const { index } = li;
+    tasks[index - 1].description = e.target.innerText;
+    saveTasks();
+  });
+
+  const editButton = li.querySelector('.edit-task');
+  editButton.addEventListener('click', () => {
+    const { index } = li;
+    const newDescription = prompt(
+      'Enter new task description',
+      tasks[index - 1].description,
+    );
+    if (newDescription) {
+      tasks[index - 1].description = newDescription;
+      li.querySelector('.task-description').innerText = newDescription;
+      saveTasks();
+    }
+  });
+}
+
+export function updateIndex() {
+  tasks.forEach((task, index) => {
+    task.index = index + 1;
+  });
+}
```

Comment:
- [ ] As requested by the [previous reviewer](https://github.com/Mekuseo/TodoList/pull/2#discussion_r1073361332), kindly ensure that when someone deletes a todo the index of the remaining todos is also updated and 1 is deleted from their index. Eg if you have 3 todos and you delete the first todo, the id of the last todo needs to be 2 and not 3. Kindly implement this.

[Screencast from 18-01-2023 04:40:37 ALASIRI.webm](https://user-images.githubusercontent.com/86318284/213186503-e7988387-bef5-4deb-bfa2-2ebecf97c827.webm)

**Note** In the screencast, I created 3 todos meaning that there will be id with indexes 1, 2, and 3. I deleted the first todo and now the id of the remaining todos should have been updated and now have id with indexes 1 and 2. Hope this is clear.

Reply:
index updated has been implemented. Thank you for your reviews